OpenRA.Mods.Common/Projectiles/AreaBeam.cs

```diff
@@ -137,6 +138,9 @@ public AreaBeam(AreaBeamInfo info, ProjectileArgs args, Color color)
 			target += dir * info.BeyondTargetRange.Length / 1024;
 
 			length = Math.Max((target - headPos).Length / speed.Length, 1);
+
+			var rangeModifiers = args.SourceActor.TraitsImplementing<IRangeModifier>().Select(a => a.GetRangeModifier()).ToArray();
```

Comment:
```suggestion
			var rangeModifiers = args.SourceActor.TraitsImplementing<IRangeModifier>().Select(a => a.GetRangeModifier());
```
No need to ToArray this if we only use the enumeration once.